homeassistant/components/yale_smart_alarm/config_flow.py

```diff
@@ -1,14 +1,32 @@
 """Adds config flow for Yale Smart Alarm integration."""
 from __future__ import annotations
 
+from typing import Any
+
+import requests
 import voluptuous as vol
 from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
 
-from homeassistant import config_entries
-from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.config_entries import (
+    CONN_CLASS_CLOUD_POLL,
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_CODE, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
 import homeassistant.helpers.config_validation as cv
 
-from .const import CONF_AREA_ID, DEFAULT_AREA_ID, DEFAULT_NAME, DOMAIN, LOGGER
+from .const import (
+    CONF_AREA_ID,
+    CONF_LOCK_CODE_DIGITS,
+    DEFAULT_AREA_ID,
+    DEFAULT_LOCK_CODE_DIGITS,
+    DEFAULT_NAME,
+    DOMAIN,
+    LOGGER,
```

Comment:
Having a local _LOGGER instance with the name of the file seems preferred to using a common logger across multiple files, and is much more common in home assistant. Perhaps for a future PR.

Reply:
I see LOGGER = logging.getLogger(__name__) in .const.py which seems to be what most other integrations are using just in a different spot.

Reply:
I see it in 22 places:
https://github.com/home-assistant/core/search?q=LOGGER+filename%3Aconst.py

Not doing it this way is in 1.7k places:
https://github.com/home-assistant/core/search?q=_LOGGER+-filename%3Aconst.py

Reply:
Should not matter

Reply:
log messages still work, yes, this is just confusing I'd you're trying to enable logging for a file or package.  It's using __name__ but most packages that do this at least use __package__ which at least makes configuring logging just appear to come from the module.  As is this is effectively logging with the wrong package name.


I realize it's not the point of this change. Definitely not following the convention of the rest of the code base at a minimum, and otherwise definitely misleading.

Am I off base here or are there advantages to  logging the incorrect file name? Again I realize it's not the scope of this pr so I suggested fixing in a follow up.

Reply:
Didn't completely get what you meant before but now I see it. Will change to package in separate PR.